src/mscorlib/src/System/Text/EncodingForwarder.cs

```diff
@@ -192,6 +192,358 @@ public unsafe static int GetBytes(Encoding encoding, char* chars, int charCount,
             return encoding.GetBytes(chars, charCount, bytes, byteCount, encoder: null);
         }
 
+#if !BIGENDIAN
+        // Ascii fast-paths
+        public unsafe static byte[] GetBytesAsciiFastPath(Encoding encoding, String s)
+        {
+            // Fast path for pure ASCII data for ASCII and UTF8 encoding
+            Debug.Assert(encoding != null);
+            if (s == null)
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.s, ExceptionResource.ArgumentNull_String);
+            Contract.EndContractBlock();
+
+            int charCount = s.Length;
+
+            byte[] bytes;
+            if (charCount > 0) {
```

Comment:
note that my comment apply in different places so I'll not repeat the comment again assuming you'll fix all touched places.

---
In reply to: [97451474](https://github.com/dotnet/coreclr/pull/8969#discussion_r97451474) [](ancestors = 97451474)